bench/cpython/optimized/unicodeobject.ll

```diff
@@ -7890,9 +7890,10 @@ if.then23:                                        ; preds = %if.then10
   ]
 
 land.lhs.true33:                                  ; preds = %if.then23
-  %spec.select.sroa.sel91.v = select i1 %cmp25, i64 5, i64 4
-  %spec.select.sroa.sel91 = getelementptr inbounds i8, ptr %buflower, i64 %spec.select.sroa.sel91.v
-  %10 = load i8, ptr %spec.select.sroa.sel91, align 1
+  %.sroa.gep110 = getelementptr inbounds i8, ptr %buflower, i64 5
+  %.sroa.gep110.val = load i8, ptr %.sroa.gep110, align 1
+  %incdec.ptr.val116 = load i8, ptr %incdec.ptr, align 1
+  %10 = select i1 %cmp25, i8 %.sroa.gep110.val, i8 %incdec.ptr.val116
```

Comment:
Regression

Reply:
Per https://github.com/llvm/llvm-project/blob/b0c6fc81fe132f20eed38bd836620dfcb5ac17e9/llvm/lib/Transforms/InstCombine/InstCombineLoadStoreAlloca.cpp#L1058 this is the preferred form right now. But maybe we should change it, now that other places have become better at handling load(select) (e.g. it has GVN support now).